### PR TITLE
Affine mapper

### PR DESCRIPTION
**Pull request by max-collins**
## Linked Issue
Fixes # (issue number)

## Implementation Checklist
- [x] My class inherits from `PolynomialBasis` (if applicable).
- [x] I have implemented all abstract methods (`evaluate_basis`, `fit`, etc.).
- [x] I have used `numpy` vectorization to avoid inefficient Python loops.
- [x] Documentation strings (docstrings) are provided for all public methods.

### 1. Mathematical Integrity
- [ ] **Convergence:** Does the implementation maintain expected accuracy? (Checked for Runge's phenomenon or stability).
- [ ] **Edge Cases:** Does it handle single points or non-equally spaced nodes correctly?
- [ ] **Domain Mapping:** Does the class correctly map the input domain to the basis domain (e.g., [-1, 1] for Chebyshev)?

### 2. API & Style
- [ ] **Method Signatures:** Methods match the Abstract Base Class exactly.
- [ ] **Naming:** Follows `CamelCase` for classes and `snake_case` for functions/variables.
- [ ] **Clean Code:** No "magic numbers"; constants are named and explained.

### 3. Testing Quality
- [ ] **Coverage:** Unit tests cover typical usage and edge cases.
- [ ] **Precision:** Tests use `np.testing.assert_allclose` instead of `==`.
- [ ] **Validation:** There is a test proving the basis can exactly reconstruct a polynomial of its own degree.

## Screenshots / Plots

### CODE REVIEW

File: src/numanalysislib/basis/affine.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        Initializes class mapping between [a,b] and [hat_a, hat_b]
+
+        Inputs:
+        basis: the polynomial basis to use
+        a: left endpoint of physical interval
+        b: right endpoint of physical interval
+        """
+        # check for endpoint ordering
+        if a > b:
+            raise ValueError("a must be greater than b")
```
Comment (wdwatson2) [inline src/numanalysislib/basis/affine.py]: "b must be greater than a"  is what you probably meant.

File: tests/test_affine.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    np.testing.assert_allclose(mapped_reference_int, physical_int)
+
+def test_failure():
+    """
+    Check for failure
+    """
+    a = -2
+    b = -4
+
+    with pytest.raises(ValueError, match = "a must be greater than b"):
```
Comment (wdwatson2) [inline tests/test_affine.py]: will need to change here too

**Review by wdwatson2 — CHANGES_REQUESTED** [review]

**Review by wdwatson2 — APPROVED** [review]
## Peer Review Checklist (To be completed by Reviewer)

### 1. Mathematical Integrity
- [x] **Convergence:** Does the implementation maintain expected accuracy? (Checked for Runge's phenomenon or stability).
- [x] **Edge Cases:** Does it handle single points or non-equally spaced nodes correctly?
- [x] **Domain Mapping:** Does the class correctly map the input domain to the basis domain (e.g., [-1, 1] for Chebyshev)?

### 2. API & Style
- [x] **Method Signatures:** Methods match the Abstract Base Class exactly.
- [x] **Naming:** Follows `CamelCase` for classes and `snake_case` for functions/variables.
- [x] **Clean Code:** No "magic numbers"; constants are named and explained.

### 3. Testing Quality
- [x] **Coverage:** Unit tests cover typical usage and edge cases.
- [x] **Precision:** Tests use `np.testing.assert_allclose` instead of `==`.
- [x] **Validation:** There is a test proving the basis can exactly reconstruct a polynomial of its own degree.

## Screenshots / Plots

File: src/numanalysislib/basis/power.py [hunk 3]
```diff
@@ -9,6 +9,11 @@ class PowerBasis(PolynomialBasis):
     """
     Implements the standard Monomial Basis: {1, x, x^2, ..., x^n}.
     """
+    def __init__(self, degree: int):
+        super().__init__(degree, a = 0, b = 1)
```
Comment (mtezzele) [inline src/numanalysislib/basis/power.py:13]: When you assign the values to function's arguments there should not be any spaces around the = sign. Please fix all the occurrences

File: tests/test_affine.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    """
+    Test if the interval [-1, 1] is successfully mapped to [-4, 5]
+    """
+    # define endpoints
+    a = -4
+    b = 5
+
+    # make instance
+    basis = PowerBasis(10)
+    Affine = AffinePolynomialBasis(basis, a = a, b = b)
```
Comment (mtezzele) [inline tests/test_affine.py]: The name of variables should not have any capital letters. Capitals are for classes.

File: tests/test_affine.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    # verify push forward maps interval
+    physical_int = np.linspace(a, b, 100)
+    reference_int = np.linspace(0, 1, 100)
+
+    mapped_reference_int = Affine.push_forward(reference_int)
+
+    np.testing.assert_allclose(mapped_reference_int, physical_int)
+
+def test_failure():
```
Comment (mtezzele) [inline tests/test_affine.py]: be more specific in the name of the test. You never know how many tests there will be so you want to provide the user with as much information as possible in the case a test fails.

**Review by mtezzele — COMMENTED** [review]
Please post here a figure to showcase the plotting works. Add some tests covering all the implemented methods and test the plotting as well.

**Comment by mtezzele** [comment]
@max-collins all the classes and methods in a test module must have `Test`/`test` as prefix in order to be discovered by `pytest`
